wrangler-service/src/main/java/co/cask/wrangler/service/explorer/FilesystemExplorer.java

```diff
@@ -210,17 +201,25 @@ private void loadFile(HttpServiceResponder responder, String scope, String path,
       properties.put(PropertyIds.SAMPLER_TYPE, SamplingMethod.NONE.getMethod());
       Format format = type == DataType.BINARY ? Format.BLOB : Format.TEXT;
       properties.put(PropertyIds.FORMAT, format.name());
-      ws.writeProperties(id, properties);
+      WorkspaceMeta workspaceMeta = WorkspaceMeta.builder(id, name)
+        .setScope(scope)
+        .setProperties(properties)
+        .build();
+      ws.writeWorkspaceMeta(workspaceMeta);
+
+      stream = new BufferedInputStream(location.getInputStream());
```

Comment:
should we use try with resources instead? Also should we move variable declaration closer to its usage?

Reply:
fixed